tests/snmp/memory.py

```diff
@@ -1,7 +1,38 @@
 #!/usr/bin/python
 
 import time
-load = []
-for i in range(0, 1000):
-    load.append([' ' * 512000])
-    time.sleep(0.01)
+import subprocess
+
+
+def get_free_memory():
+    command = "grep MemFree /proc/meminfo | awk '{print $2}'"
+    output = subprocess.check_output(command, shell=True)
+    free_memory = int(output.strip())
+    return free_memory
+
+
+free_memory = get_free_memory()
+total_chars = 512000000
+
+if (free_memory * 1024) > int(total_chars * 130 / 100):
```

Comment:
Can you elaborate on why you are checking against (total_chars * 130 / 100)?

Reply:
Thanks for reviewing @SuvarnaMeenakshi , checking the free memory here is for avoiding memory corruption.
If free memory is smaller than 130% percent of total_chars, then use chunk memory instead.
130% here is only for reserving some buffer. thanks